demo/app.py

```diff
@@ -23,6 +28,33 @@ def get_access():
 
 @app.route('/', methods=['GET', 'POST'])
 def index():
+    cas_url = app.config['CAS_URL']
+    service_url = app.config['SERVICE_URL']
+    authorized_users = app.config['AUTHORIZED_USERS']
+    logout_url = service_url+"logout"
+
+    if u'ticket' in request.args:
```

Comment:
Just curious, is there any particular reason explicitly encode the string as unicode string? Is there any non-ASCII character we will get from `request.args`?

Reply:
Good question. I will defer to @magical 's judgment since he originally wrote this.

Reply:
Good question. I know Flask usually decodes strings into unicode whenever possible, so i was probably worried that looking for a non-unicode `'ticket'` would either 1) silently fail or 2) cause a UnicodeDecodeError if there happened to be a non-ascii value in `request.args`. However, neither of those seem to be an issue.

```
% cat request-type.py 
from flask import Flask, request

app = Flask(__name__)

@app.route('/')
def index():
    body = [
        repr('ticket' in request.args),
        repr(u'ticket' in request.args),
        repr(request.args)
    ]
    return "\n".join(body), 200, {'content-type': 'text-plain'}

if __name__ == '__main__':
    app.run()
% python request-type.py&                 
[1] 26480
%  * Running on http://127.0.0.1:5000/ (Press CTRL+C to quit)

% curl 'http://localhost:5000/?ticket=1&tíçkët=2'
127.0.0.1 - - [02/Mar/2018 12:24:46] "GET /?ticket=1&tíçkët=2 HTTP/1.1" 200 -
True
True
ImmutableMultiDict([(u't\xed\xe7k\xebt', u'2'), ('ticket', u'1')])
```

Not only does python not seem to care what type `'ticket'` is, flask doesn't even decode all-ascii request args!